test/tests/functional/pbs_test_run_count.py

```diff
@@ -123,16 +145,87 @@ def test_run_count_subjob(self):
         a = {ATTR_J: '1-2'}
         j = Job(TEST_USER, a)
         jid = self.server.submit(j)
+        self.subjob_check(jid=jid, sjid=j.create_subjob_id(jid, 1))
 
-        self.server.expect(JOB, {ATTR_state: "H", ATTR_runcount: 21},
-                           attrop=PTL_AND, id=j.create_subjob_id(jid, 1))
+    def test_run_count_subjob_in_x(self):
+        """
+        Submit a job array and check if the subjob and the parent are getting
+        held after 20 rejection from mom when there is another subjob in X
+        """
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 1},
+                            id=self.mom.shortname)
 
-        ja_comment = "Job Array Held, too many failed attempts to run subjob"
+        # Create an execjob_begin hook that rejects the job
+        self.create_reject_begin_hook()
+        time.sleep(2)
+        self.disable_reject_begin_hook()
 
-        self.server.expect(JOB, {ATTR_state: "H", ATTR_comment: (MATCH_RE,
-                           ja_comment)}, attrop=PTL_AND, id=jid)
+        a = {ATTR_J: '1-6'}
+        j = Job(TEST_USER, a)
+        j.set_sleep_time(10)
+        jid = self.server.submit(j)
+        time.sleep(9)
+        self.server.expect(JOB, {ATTR_state: "R"},
+                           id=j.create_subjob_id(jid, 2))
+        self.enable_reject_begin_hook()
+        time.sleep(8)
+        self.server.expect(JOB, {ATTR_state: "X"},
+                           id=j.create_subjob_id(jid, 2))
+
+        self.subjob_check(jid=jid, sjid=j.create_subjob_id(jid, 3))
+
+    def test_large_run_count_subjob(self):
+        """
+        Submit a job array with a large (>20) but valid run_count value and
+        check if the subjob and the parent are getting
+        held after 1 rejection from mom
```

Comment:
> held after 1 rejection from mom

This is confusing. Did you meant to say "after first job getting rejected run_count times?"

Reply:
its clear, i mean "one" rejection not first. "1st" means first

Reply:
But does that supposed to get held after "1" rejection or maxruncount rejections?

Reply:
This is the behavior for all jobs when run_count is more than 20. May be you are not familiar with this. please read through docs or experiment on product before commenting.